### ISSUE TITLE
Membership proposal process of Tomas Ives vanishes

### ISSUE THREAD

**Issue by brunobettati** [deora-earth/rosebudDAO#53]
<img width="2560" alt="Screen Shot 2020-03-30 at 20 50 43" src="https://user-images.githubusercontent.com/59749599/77950034-3b2c3500-72c8-11ea-9f1d-8499ad37212b.png">

During the "review" period the membership proposal of tomas ives disappeared and reappeared two times at least. I now went into the site and it is there, "ready for processing." (see scr capture)

<img width="2560" alt="Screen Shot 2020-03-30 at 20 52 56" src="https://user-images.githubusercontent.com/59749599/77950204-73337800-72c8-11ea-8c12-e181f51eba11.png">
<img width="2560" alt="Screen Shot 2020-03-30 at 20 53 12" src="https://user-images.githubusercontent.com/59749599/77950240-7e86a380-72c8-11ea-8732-1083ee647e45.png">
<img width="2560" alt="Screen Shot 2020-03-30 at 20 53 48" src="https://user-images.githubusercontent.com/59749599/77950293-92caa080-72c8-11ea-9bb9-879a08fd6bad.png">

But during the transaction with wallet to pay the reward, the system throws error. 
As i try to process reward again, it now throws constant error message.

I logged out and then back in. The proposal appears now in "complete", correctly processed. But the reward is NOT in the wallet.

I am sorry to notify that membership proposal, a key process for the start up of the Fund and for its launch, is plagued by bugs which are very confusing for an unexperienced user. Thus i wiill create a "milestone."
<img width="2560" alt="Screen Shot 2020-03-30 at 21 00 57" src="https://user-images.githubusercontent.com/59749599/77950931-93176b80-72c9-11ea-8dc0-13976c2332ff.png">
<img width="2560" alt="Screen Shot 2020-03-30 at 21 01 26" src="https://user-images.githubusercontent.com/59749599/77950981-a1fe1e00-72c9-11ea-8fbf-0e51c2143fa5.png">

**Comment by MaxKaay** [comment]
What error messages did you see here? Can you remember?

**Comment by brunobettati** [comment]
I have sorted out this bug myself. And it is related to issues #20 #21 #23 and #47 

Rosebud Fund has 4 types of transactions which involve GAS (in Eth) and also DAI: 
guarantees ("PROCESS PROPOSAL" in Metamask)
rewards (again "PROCESS PROPOSAL" in Metamask)
tributes (i think it is "CONTRACT INTERACTION" in Metamask) 
guarantee devolution ((i think it is "CONTRACT INTERACTION" in Metamask)  

All other transactions (submit, vote, approve Rosebud) only charge Gas (and appear under the form of "APPROVE" or "SUBMIT VOTE" in Metamask.

What i figured out, is that after you operate the fund, some functions do not proceed entirely unless **user logs-out and then logs-in again with his wallet.** This might explain the "disappearance" of the voting of Tomas, and many other issues. 

Last nite I tested UX by sending my proposal for Alan's  membership.  i noticed that due to lack of pre-approved tribute (Alan had 5 DAI), submission didn't proceed, although i previously ensured Alan went to DAI center and clicked on "Approve Rosebud", we then went to approve the submission and agin it did not work.  

Going back to DAI Center, i noticed that in gray, the 100 DAI did not figure there, only 5. But the 100 DAI were already vanished from Alan's wallet. So inspired by the "refresh" function of browsers, i suggested Alan to log out the wallet out and in again, and poof! now the DAI approved went from 5 to 105.

But the approved proposal did not show up at "in queue" (and so, it disappeared). I guessed that we had to wait the "Period" (a fixed parameter of 2.4 hours) to see it show up at "voting period." So it did. I just voted.

From what I remember of Tomas membership Proposal, I can predict that Alan's proposal will also vanish during the "Grace Period." This would explain the second disapperance. 

For the sake of clarity and transparency, it would be good that the transactional names of Metamask match the transactions performed by Rosebud. The transaction called "Approved" is particularly ambigous.

And concerning log-in or log-out of wallet, we must display a message somewhere on the likes of _"If your transaction does not appear on your wallet, or you receive an error message, please try logging out of Rosebud and then logging in your wallet again. If the problem persists, please email contact@rosebud.fund."_   (and i don't agree this should have an extra cost, specially if transactions present such lack of clarity to the point the we might confront criticism and lack of confidence on the security of transactions, a key point that will undermine the whole credibility of the project.)

The mail contact@rosebud.fund is operational and i have asked in a previous issue who will do maintenance to forward it in cases like this one. I can also prepare a specific transactions@rosebud.fund for users who specifically have a problem regarding a transaction, and leave it on automatic forwarding to the person in charge of maintenance. 

**Please note that this observation does not include the "reward disappearance" i spotted and commented on the milestone.**

**Comment by johba37** [comment]
> But the approved proposal did not show up at "in queue" (and so, it disappeared). I guessed that we had to wait the "Period" (a fixed parameter of 2.4 hours) to see it show up at "voting period." So it did. I just voted.

yeah, that is the interference i was talking about. agree that it needs to be fixed.

> For the sake of clarity and transparency, it would be good that the transactional names of Metamask match the transactions performed by Rosebud. The transaction called "Approved" is particularly ambigous.

the description in metamask are pulled from popular examples of functionnames accross contracts on Ethereum. to my knowledge, we can not affect those.

> and i don't agree this should have an extra cost, specially if transactions present such lack of clarity to the point the we might confront criticism and lack of confidence on the security of transactions, a key point that will undermine the whole credibility of the project.

i have not observed such behaviour when i tested the prototype, hence we should work on reproducing the bug. then we can decide on pricing. 

overall i'm happy to see that there are 100$ in the bank now. also a confirmation that the delivered contracts are doing their jobs 💪

**Comment by johba37** [comment]
i have a fix in tests, confirmation pending
